### ISSUE TITLE
Move Loki to Plugin

### ISSUE THREAD

**Issue by thombruce** [thombruce/souroranges#8]
At present, LokiJS is invoked in one place: the `entities()` store. Since it will need to be used in other stores, we want to migrate it into its own general purpose plugin.

I've been banging my head against this problem for a bit. Most of it's trivial, but I'm struggling to pass the db to the store once invoked.

Ideally, the database would be setup during the app initialisation. Following this, views/components should be able to request data via Pinia which invokes the same database instance (loading a collection from file if it has to).

Since only Pinia makes use of Loki, it made sense to me to try and make this a Pinia plugin (I've also tried a general Vue plugin approach). The difficulty with both is passing the database back to the store module.

---

As a first step, _maybe_ we can just generalise the code **as is**. Meaning move the Loki logic into its own file and invoke it in exactly the same way. Simply cleaning up the Entities store file.

Then, if/when it's beneficial to try and work out when this logic _ought_ to be executed, we can take another look at that in the future.

Any potential flaws with this approach? _Possibility of creating/exporting multiple databases? This is avoided by the singleton pattern, no?_

**Comment by thombruce** [comment]
I can get this 99% of the way there in a heartbeat. The problem is `initStore()`. This _was_ handled by a callback after the database had loaded.

I've tried invoking it directly in the store, as a sort of beforeAction at the start of other actions (with a `dbLoaded` boolean determining whether or not to run through it), and in the setup script on a component.

_As a beforeAction on other actions was a silly idea, by the way; these actions aren't invoked until we attempt to add/remove data, so of course they didn't succeed in an initial load._

Perhaps the issue is that setup is too early. Yeah, fair enough. I get really bogged down trying to keep everything super sleek. I think it's fine to invoke the data load later in the process when we _know_ the database is ready and loaded.

**Comment by thombruce** [comment]
The problem is this.... Loading the database is asynchronous. It can trigger a callback, which is great, but we don't necessarily want that callback to be specific to any particular view/component/store. It ought to be generalised such that it shouldn't matter where you load into the app.

The callback could trigger another function which is somehow aware of the current page state and loads the appropriate collection(s)/dynamicView(s) based on that.

To achieve that... I think _for now_ we'll simply invoke that same function we already have. It's specific to the one store, but it exists and if we can get it working we've made a crucial step forwards.

After that, I see two options:

1. A root store that is "aware" of which child store needs to be loaded and redirects the callback. _Not ideal; introduces new, unnecessary state and mutations to provide said "awareness"._
2. Vue Router. We're gonna want it anyway, may as well use the route information to determine which content needs to be loaded. This is just the sensible option in my eyes, making secondary use of something we've already determined we'll be using. _I've gotta imagine that if we already had Vue Router in place, I might not be banging my head against this problem quite so much._
